doc/whats-new.rst

```diff
@@ -138,6 +138,10 @@ Bug fixes
 - Fixed sub-optimal performance in certain operations with object arrays (:issue:`1121`).
   By `Yves Delley <https://github.com/burnpanck>`_.
 
+- Use `safe_cast_to_index` when generating `unique_value_groups` in ``groupby``
```

Comment:
Can you describe this in terms of what it fixes rather than how your fix works? E.g., "fixed a bug with grouping by datetime variables"